src/main.cpp

```diff
@@ -5572,19 +5600,17 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
         if (pfrom->fWhitelisted && GetBoolArg("-whitelistrelay", DEFAULT_WHITELISTRELAY))
             fBlocksOnly = false;
 
-        LOCK(cs_main);
-
         for (unsigned int nInv = 0; nInv < vInv.size(); nInv++)
         {
-            const CInv &inv = vInv[nInv];
-
             boost::this_thread::interruption_point();
 
-            bool fAlreadyHave = AlreadyHave(inv);
-            LOG(NET, "got inv: %s  %s peer=%d\n", inv.ToString(), fAlreadyHave ? "have" : "new", pfrom->id);
-
+            const CInv &inv = vInv[nInv];
             if (inv.type == MSG_BLOCK)
```

Comment:
Shouldn't we add also `... && (inv.type == MSG_XTHINBLOCK) && (inv.type == MSG_THINBLOCK)` to the `if` condition?

Reply:
No I don't so.. the caller will know if it's a block or txn inventory, and it doesn't matter if its a BLOCK/THINBLOCK/XTHIN here, we only use  the inv to get the hash...probably would be clearer now to get rid of passing the inv and just use the hash.

Reply:
Sorry, I completely misread your comment.  Yeah, here there is no possibility of thinblock or xthinblock inv types here...we never actually send or received an INV for a thinblock/xthin.  We only use such a INV for sending the get_xthin message.

Reply:
You bring a good point here...we should really check specifically for a TX inventory type and then if we don't get either the block or tx inv then we should probably give them a DOS () for sending us invalid inv types.  But that's for another PR.  I"ll make a note of it.

Reply:
opened PR #1084 for this issue.